e2e/helpers/graphs.ts

```diff
@@ -0,0 +1,34 @@
+/*global page*/
+
+import { host } from '../config';
+import { toMatchImageSnapshot } from 'jest-image-snapshot';
+
+expect.extend({ toMatchImageSnapshot });
+
+export async function displayGraph(type: string) {
+  const pageID = page.url().split("/").pop();
+  const pageUrl = `${host}/page/` + pageID;
+
+  await expect(page).toClick('a', { text: '(view page)' });
+
+  //waits until the target is available [see browser.targets]
+  const graphsPageTarget = await browser.waitForTarget(target => target.url() === pageUrl);
+  const graphsPage = await graphsPageTarget.page();
+  await graphsPage.waitFor(2000); // wait for the chart visualization animations to end
+  
+  let chartContainer: HTMLElement | any;
+  if(type === 'list'){
+    chartContainer = await graphsPage.$('.ListChart ');
+  } else {
+    chartContainer = await graphsPage.$('.recharts-responsive-container');
+  }
+  
+  const chartScreenshot = await chartContainer.screenshot();
+  expect(chartScreenshot).toMatchImageSnapshot({
+    failureThreshold: 0.03,
+    failureThresholdType: 'percent',
+    allowSizeMismatch: true,
+  });
+
+  await graphsPage.close();
```

Comment:
@elric-wamugu i like the idea of abstracting the graph rendering, but can you move pls the testing part to each of the test ? this way you can see the expect on the test itself and you can avoid the conditional for the type.